Cleaning Log I

Import libraries (panda & seaborn)

load .csv file containing data and store in variable, 'df'

In [11]:
import pandas as pd
import seaborn as sns

df = pd.read_csv("usarrests.csv")

Cleaning Log II

dispay the number of rows & columns, given by the .shape() function

In [12]:
df.shape

(50, 4)

Cleaning Log III

dispay the first 10 rows, given by the .head(10) function

In [13]:
df.head(10)

,Unnamed: 0,Murder,Assault,UrbanPop
0,Alabama,13.2,236.0,58
1,Alaska,10.0,263.0,48
2,Arizona,8.1,294.0,80
3,Arkansas,8.8,190.0,50
4,California,9.0,276.0,91
5,Colorado,7.9,204.0,78
6,Connecticut,3.3,110.0,77
7,Delaware,5.9,238.0,72
8,Florida,15.4,335.0,80
9,Georgia,17.4,NaN,60


There is at least one problems that I can already see in this data:

There is at least one rows, that are either blank entries or characters not representing a number, in the Assault column for Georgia. This values is represented by "NaN".

Cleaning Log IV

dispay the data types of the columns, given by the .dtypes() function

In [14]:
df.dtypes

Unnamed: 0        str
Murder        float64
Assault       float64
UrbanPop        int64
dtype: object

Cleaning Log V

dispay the blank entries of the columns, given by the .isna().sum() function

In [15]:
df.isna().sum()

Unnamed: 0    0
Murder        0
Assault       1
UrbanPop      0
dtype: int64

Cleaning Log VI

1. Fill the missing Assault value using the 'median' statistic that resists skewed numbers of Assaults in GA.
2. Print the fill value I used and record the fill value for your cleaning log.
3. I will show that there are 0 missing prices afterwards.

In [16]:
fill_value = df["Assault"].mean()
print(f'The fill value represented by the mean of the price column is {fill_value}.')
df["Assault"] = df["Assault"].fillna(fill_value)
df.isna().sum()

The fill value represented by the mean of the price column is 182.18367346938774.


Unnamed: 0    0
Murder        0
Assault       0
UrbanPop      0
dtype: int64

Cleaning Log VII

Here, I show that there were 0 duplicates in the data frame before the .drop_duplicates() function. 

In [17]:
print(df.duplicated().sum())
df = df.drop_duplicates()

0


Cleaning Log VIII

Via the shape() function, I now show the quantity of rows of the data frame are now 50. Same as it has always been even with the data cleaning we have done thus far.

In [18]:
df.shape

(50, 4)

Cleaning Log IX

compute summary statistics usoing the describe() function.

In [19]:
df.describe()

,Murder,Assault,UrbanPop
count,50.00000,50.000000,50.00000
mean,7.78800,182.183673,74.20000
std,4.35551,129.535067,73.40828
min,0.80000,45.000000,6.00000
25%,4.07500,109.000000,53.25000
50%,7.25000,159.000000,66.00000
75%,11.25000,249.000000,77.75000
max,17.40000,879.000000,570.00000


Cleaning Log X

Change Log Summary

The one maximum is far outside any plausible range is Iowa with a value of 570. This scews the data when accessng patters and trends in the data for that column and all other data processig that involves Iowa's row; this value is extreme. GA's missing data in the assault column was replaced by a value computed by the other colummns' mean value in the table. This was plausible when considering neighboring values when the data was sorted by UrbanPop. The median value for that field would have been outside the range of acceptable values when considering neighboring trends for the data.

https://github.com/chillyvee70/cs82a-portfolio/tree/main/module_3